src/xai_grok/grok.py

```diff
@@ -0,0 +1,223 @@
+import http
+import logging
+
+import pydantic
+import requests
+
+import xai_grok.errors as errors
+import xai_grok.providers as providers
+import xai_grok.schemas as schemas
+
+
+class Grok:
+
+    GROK_API_BASE_URL = "https://api.x.ai"
+
+    logger = logging.getLogger(__name__)
+
+    def __init__(
+        self,
+        api_key: str,
+        requests: requests,
```

Comment:
The `http.HTTPMethod` is incorrectly used, which will cause an `AttributeError` since `http` module does not have `HTTPMethod`. Use `requests` methods directly instead.